delegator.py

```diff
@@ -107,14 +108,15 @@ def return_code(self):
     def std_in(self):
         return self.subprocess.stdin
 
-    def run(self, block=True):
+    def run(self, block=True, binary_stdio=False):
         """Runs the given command, with or without pexpect functionality enabled."""
         self.blocking = block
 
         # Use subprocess.
         if self.blocking:
-            s = subprocess.Popen(self._popen_args, **self._default_popen_kwargs)
-
+            popen_kwargs = self._default_popen_kwargs
+            popen_kwargs['universal_newlines'] = not binary_stdio
```

Comment:
So this approach has some subtle implications. Without calling `.copy()` on `self._default_popen_kwargs`, you're mutating the object's dictionary here too. That means a future call with `blocking=False` may have unexpected kwargs.

Could you also clarify why toggling `universal_newlines` is the desired solution here? This seemed to be an issue with Python 2 v Python 3 when we discussed it in IRC. I'm a little apprehensive about adding additional parameters if we don't need them.